### DISCUSSION TITLE
Directx12 screenshot not supported

### DISCUSSION THREAD

**Discussion by monkeycc** [mhammond/pywin32#2540] [General]
```
    hWnd = win32gui.FindWindow(0,winwin)  
    left, top, right, bot = win32gui.GetWindowRect(hWnd)
    width = right - left
    height = bot - top
    hWndDC = win32gui.GetWindowDC(hWnd)
    mfcDC = win32ui.CreateDCFromHandle(hWndDC)
    saveDC = mfcDC.CreateCompatibleDC()
    saveBitMap = win32ui.CreateBitmap()
    saveBitMap.CreateCompatibleBitmap(mfcDC, width, height)
    saveDC.SelectObject(saveBitMap)
    saveDC.BitBlt((0, 0), (width, height), mfcDC, (0, 0), win32con.SRCCOPY)
```
Directx12 screenshot not supported

Black screen

**Comment by mhammond** [comment]
What makes you think that's expected to work? Does the equivalent C++ code work? I suspect this isn't a bug in pywin32, but instead a misunderstanding of how it should work.

**Comment by monkeycc** [comment]
> What makes you think that's expected to work? Does the equivalent C++ code work? I suspect this isn't a bug in pywin32, but instead a misunderstanding of how it should work.

Thank you for your reply


World of Warcraft - window mode

Graphics settings - directx12

Screenshot failed

I don't know how to solve the problem

I don't know the principle

The Win32 interface does not support directx12?
Direct3D 12 header and library are windows 10 SDK 
Windows 10 already includes DirectX 12.

**Comment by mhammond** [comment]
I don't know how to take directx screenshots - but this is a support request, not a bug report, so please see https://github.com/mhammond/pywin32#support

**Comment by Avasam** [comment]
There's a ton of reasons BitBlt screenshot method may not work:
- exclusive fullscreen application
- Odd rendering pipeline
- OpenGL
- Hardware acceleration
- UWP app
- ...

I'd suggest just using a different screen recording/screenshot method, like:
- Windows' Graphic Capture API, aka WGC, available on Windows 10+
- DirectX Desktop Duplication
